lib/topological_inventory/orchestrator/health_check.rb

```diff
@@ -0,0 +1,62 @@
+module TopologicalInventory
+  module Orchestrator
+    class HealthCheck
+      include Logging
+
+      attr_accessor :sources, :topology, :k8s
+
+      def initialize(sources, topology, k8s)
+        @sources = sources
+        @topology = topology
+        @k8s = k8s
+      end
+
+      def run
+        loop do
+          checks
+
+          sleep configured_seconds
+        end
+      end
+
+      def checks
+        errors = []
+
+        errors << check_url(topology)
+        errors << check_url(sources)
+        errors << check_k8s(k8s)
+
+        if errors.any?
```

Comment:
I'm wondering if it isn't too strict. 
On one side, the orchestrator is live is at least one check is true.
On the other side, the orchestrator can't work properly if it doesn't have access to all - but we're not as strict in other services

Reply:
Yeah, I'm of the opinion that if we can't hit one of the APIs then orchestrator isn't going to be functioning correctly. 

In order for the pod to restart as well it would have to fail 3 times in a row (which means for 3 minutes straight). 

I think it's reasonable, but would like to hear everyone's opinion cc @syncrou @roliveri 

---

We could potentially make it be 2+ required to mark it un-healthy etc too, or require all to go down.

Reply:
Sorry I missed the email notifications of your request.

If access to all is required for proper function, then I would think it isn't healthy if any cannot be reached.
Does the fact that some of the failures may be transient in nature come into play?

Reply:
Yes - we account for that in the OCP deployment. It checks once a minute and if it fails for 3 minutes in a row _then_ it will be restarted. That way if it fails once it is just a blip.

Reply:
So let's keep it as is

Reply:
I think so. It make sense to me.